server/service/osquery_utils/queries.go

```diff
@@ -503,14 +510,13 @@ SELECT
   '' AS vendor,
   '' AS arch
 FROM python_packages;
-`,
+`),
 	Platforms:        fleet.HostLinuxOSs,
 	DirectIngestFunc: directIngestSoftware,
 }
 
 var softwareWindows = DetailQuery{
-	Query: `
-WITH cached_users AS (SELECT * FROM users WHERE directory <> '')
+	Query: withCachedUsers(`WITH cached_users AS (%s)
```

Comment:
We need to keep the `directory <> ''` on this one.

Reply:
Do you remember why? Do we want that in the other users too?

Reply:
https://github.com/fleetdm/fleet/commit/bc2137e1327e5191a006d3b42337babcc000c61a

We should add what Zach there as a code comment for future reference.

Reply:
+1, I believe it makes sense to add it to all the software queries.

Reply:
I think it probably makes sense to filter out users without home directories in most of the host vitals queries.